app/src/main/java/com/cornellappdev/coffee_chats_android/CreateProfileActivity.kt

```diff
@@ -0,0 +1,140 @@
+package com.cornellappdev.coffee_chats_android
+
+import android.content.Intent
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.widget.ArrayAdapter
+import androidx.appcompat.app.AppCompatActivity
+import com.cornellappdev.coffee_chats_android.models.InternalStorage
+import com.cornellappdev.coffee_chats_android.models.UserProfile
+import kotlinx.android.synthetic.main.activity_create_profile.*
+import java.util.*
+import kotlin.collections.ArrayList
+
+
+class CreateProfileActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_create_profile)
+
+        // variables to keep track if editTexts are filled out
+        var majorFilled = false
+        var hometownFilled = false
+
+        // tries to retrieve User Profile from internal storage
+        var profile = InternalStorage.readObject(this, "profile") as UserProfile
+        demoTop.text = getString(R.string.demographics_header, profile.userName)
+        if (profile.hometown.isNotEmpty())  {
+            hometownET.setText(profile.hometown)
+            hometownFilled = true
+        }
+        if (profile.major.isNotEmpty()) {
+            majorACTV.setText(profile.major)
+            majorFilled = true
+        }
+
+        // nextButton is disabled until user has filled out all required info
+        if (!hometownFilled || !majorFilled) {
+            nextButton.isEnabled = false
+            nextButton.isClickable = false
+        }
+
+        // Initializing the class spinner
+        val classArray = ArrayList<String>()
+        var year = Calendar.getInstance().get(Calendar.YEAR)
+        val month = Calendar.getInstance().get(Calendar.MONTH)
+        if (month >= 6) year++  // sets minimum graduation year to next year
+        for (i in 0..4) {
```

Comment:
More so something to ask design about, but should we display more than just the 4 years?